File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    // Log line format: `[LEVEL]: MESSAGE`
    std::string message(std::string line) {
        int msg_start_idx = line.find(":") + 2;
        return line.substr(msg_start_idx);
    }

    std::string log_level(std::string line) {
        int close_bracket_idx = line.find("]");
        return line.substr(1, close_bracket_idx - 1);
    }
    
    std::string reformat(std::string line) {
        return message(line) + " (" + log_level(line) + ")";
    }
}  // namespace log_line

File: solutions/julia/log-levels/1/log-levels.jl
function message(msg)
    return strip(split(msg, ':')[2])
end

function log_level(msg)
    return lowercase(split(msg, ':')[1][begin + 1:end - 1])
end

function reformat(msg)
    return "$(message(msg)) ($(log_level(msg)))"
end
